python-challenges/cli-password-manager/devashmit-day33/manager.py

```
@@ -0,0 +1,51 @@
+
+VAULT=Path("vault.json")
+def derive_key(pw,salt):kdf=PBKDF2HMAC(algorithm=hashes.SHA256(),length=32,salt=salt,iterations=480_000);return base64.urlsafe_b64encode(kdf.derive(pw.encode()))
+def load():return json.loads(VAULT.read_text()) if VAULT.exists() else{"salt":None,"entries":{}}
+def save(v):VAULT.write_text(json.dumps(v,indent=2))
+def fernet(v,pw):return Fernet(derive_key(pw,bytes.fromhex(v["salt"])))
+
+def cmd_add(_):
+    v=load();pw=getpass.getpass("Master passphrase: ")
+    if v["salt"] is None:v["salt"]=os.urandom(16).hex();save(v);print("  Vault created.")
+    f=fernet(v,pw);site=input("Site: ").strip();user=input("Username: ").strip();passwd=getpass.getpass("Password: ")
+    v["entries"][site]={"username":user,"password":f.encrypt(passwd.encode()).decode()};save(v);print(f"  Saved '{site}'.")
```

Ruff (0.15.20)

[error] Multiple statements on one line (semicolon)

(E702)